rest_api/simple_supply_rest_api/database.py

```diff
@@ -13,14 +13,59 @@
 # limitations under the License.
 # ------------------------------------------------------------------------------
 
+import asyncio
+import logging
+
+import aiopg
+import psycopg2
+
+
+LOGGER = logging.getLogger(__name__)
+
 
 class Database(object):
-    """Simple object for managing a connection to a postgres database
+    """Manages connection to the postgres database and makes async queries
     """
-    def __init__(self, host, port, name, user, password):
-        self._host = host
-        self._port = port
-        self._name = name
-        self._user = user
-        self._password = password
+    def __init__(self, host, port, name, user, password, loop):
+        self._dsn = 'dbname={} user={} password={} host={} port={}'.format(
+            name, user, password, host, port)
+        self._loop = loop
         self._conn = None
+
+    async def connect(self, retries=5, initial_delay=1, backoff=2):
+        """Initializes a connection to the database
+
+        Args:
+            retries (int): Number of times to retry the connection
+            initial_delay (int): Number of seconds wait between reconnects
+            backoff (int): Multiplies the delay after each retry
+        """
+        LOGGER.info('Connecting to database')
+
+        delay = initial_delay
+        for attempt in range(retries):
```

Comment:
If I understand it right, this can be simplified as follows:

```
for x in blah:
    try:
        do_thing()  # if unsuccessful, try again
        log('succeeded')  # this only executes if the thing succeeded
        return  # we're done here
    except someerror:
        handle_error()

    # no need for else clause, since the early return will prevent any of this from executing

    do_thing()  # last attempt, come what may
    log('succeeded')  # this won't get executed if the last try failed
```